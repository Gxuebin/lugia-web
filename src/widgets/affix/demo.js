/**
 *
 * create by guorg
 *
 * @flow
 */
import * as React from 'react';
import styled from 'styled-components';
import Affix from './affix';
import Button from '../button';

export default class AffixDemo extends React.Component {
  constructor() {
    super();
    this.EleRef = null;
    this.Ele2Ref = null;
    this.state = {
      display: true,
    };
  }
  render() {
    const { display } = this.state;
    return (
      <div>
        {display ? (
          <div style={{ width: '200px', height: '1200px' }}>
            <div style={{ width: '200px', height: '200px' }} />
            <Affix offsetTop={50}>
              <Button>affix-top</Button>
            </Affix>

            <div
              style={{ width: '200px', height: '200px', overflowY: 'scroll' }}
              ref={node => (this.EleRef = node)}
            >
              <div style={{ height: '400px' }}>
                <div style={{ width: '20px', height: '100px' }} />
                <Affix offsetTop={50} target={() => this.EleRef}>
                  <Button>affix-top</Button>
                </Affix>
              </div>
            </div>
            <div
              style={{ width: '200px', height: '200px', overflowY: 'scroll' }}
              ref={node => (this.Ele2Ref = node)}
            >
              <div style={{ height: '400px' }}>
                <div style={{ width: '20px', height: '200px' }} />
                <Affix offsetBottom={50} target={() => this.Ele2Ref}>
                  <Button>affix2-bottom</Button>
                </Affix>
              </div>
            </div>

            <div style={{ width: '200px', height: '500px' }} />
            <Affix offsetBottom={50}>
              <Button>affix-bottom</Button>
            </Affix>
            <div style={{ width: '200px', height: '500px' }} />
          </div>
        ) : (
          <div style={{ width: '200px', height: '1200px' }} />
        )}
      </div>
    );
  }
}
