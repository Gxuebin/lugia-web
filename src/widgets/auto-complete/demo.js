import * as React from 'react';
import AutoComplete from './';

const data = [
  'Nikcy Romero',
  'Armin van Buuren',
  'Hardwell',
  'Zedd',
  'Kazze',

  'Vicetone',
  'Martin Garrix',
  'David Guetta',
  'The Chainsmokers',
  'Kygo',

  'Axwell ^ Ingrosso',
  'Dimitri Vegas & Like Mike',
  'Calvin Harris',
  'Avicci',
  'Fedde Le Grand',

  'Tiesto',
  'Snoop Dogg',
  'Bassjackers',
  'Sebastian Ingrosso',
  'Swedish House Mafia',
  'Alesso',
  'Afrojack',
  'Knife Party',
  'Dannic',
  'R3hab',
];

export default class AutoCompleteBounded extends React.Component<any, any> {
  constructor(props) {
    super(props);
    this.state = {
      menuData: data,
    };
  }
  static getDerivedStateFromProps(props: any, state: any) {
    const hasValueInProps = 'value' in props;
    const value = hasValueInProps ? props.value : state ? state.value : props.defaultValue;
    if (!state) {
      return {
        value,
      };
    }
    return {
      value,
    };
  }

  render() {
    const { menuData, value } = this.state;
    return <AutoComplete showOldValue value={value} data={menuData} onChange={this.onChange} />;
  }

  onChange = (value: string) => {
    this.search(value);
    this.setState({ value });
  };

  search(query: string) {
    let menuData;
    let rowSet = [];
    const len = data.length;

    for (let i = 0; i < len; i++) {
      const row = data[i];
      if (this.searchValue(query, row)) {
        rowSet.push(row);
      }
      if (query === row) {
        rowSet = [];
        break;
      }
    }

    if (rowSet.length === len) {
      menuData = data;
    } else {
      menuData = rowSet;
    }
    this.setState({ menuData });
  }

  searchValue = (query: string, row: string): boolean => {
    return row.indexOf(query) !== -1;
  };
}

// export default class Demo extends React.Component<any, any> {
//   constructor(props) {
//     super(props);
//     this.state = {
//       menuData: data,
//       value: '',
//     };
//   }
//   render() {
//     return (
//       <AutoComplete
//         data={this.state.menuData}
//         onChange={this.onChange}
//         placeholder={'请输入'}
//         showOldValue={true}
//       />
//     );
//   }

//   onChange = value => {
//     this.search(value);
//   };

//   search(query) {
//     let menuData;
//     let rowSet = [];
//     const len = data.length;

//     for (let i = 0; i < len; i++) {
//       const row = data[i];
//       if (query === row) {
//         rowSet = [];
//         break;
//       }

//       if (this.searchValue(query, row)) {
//         rowSet.push(row);
//       }
//     }

//     if (rowSet.length === len) {
//       menuData = data;
//     } else {
//       menuData = rowSet;
//     }
//     this.setState({ menuData });
//   }

//   searchValue = (query, row) => {
//     return row.indexOf(query) !== -1 || row === query;
//   };
// }
