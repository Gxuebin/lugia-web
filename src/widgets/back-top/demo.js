/**
 *
 * create by guorg
 *
 * @flow
 */
import * as React from 'react';
import styled from 'styled-components';
import BackTop from './back-top';
import Widget from '../consts/index';
import Theme from '../theme';

const Demo = styled.div`
  & > span > div > div {
    bottom: ${props => 50 + props.cur * 50}px;
  }
`;
const DemoBack = styled.div`
  width: 40px;
  height: 40px;
  line-height: 40px;
  text-align: center;
  border-radius: 4px;
  border: 1px solid #e8e8e8;
`;
export default class AffixDemo extends React.Component<any, any> {
  EleRef: any;
  render() {
    const view = {
      [Widget.BackTop]: {
        normal: {
          color: 'red',
          background: { backgroundColor: 'orange' },
          width: 100,
          height: 100,
          borderRadius: 100,
          opacity: 0.7,
          border: {
            borderWidth: 1,
            borderStyle: 'solid',
            borderColor: 'red',
          },
        },
      },
    };
    return (
      <div style={{ height: '1600px' }}>
        <BackTop />
        <Demo cur={1}>
          <BackTop>
            <DemoBack>UP</DemoBack>
          </BackTop>
        </Demo>
        <Demo cur={2}>
          <Theme config={view}>
            <BackTop />
          </Theme>
        </Demo>
        <div
          style={{ width: '200px', height: '200px', overflowY: 'scroll', marginTop: '400px' }}
          ref={node => (this.EleRef = node)}
        >
          <Demo cur={3}>
            <div style={{ width: '20px', height: '300px' }} />
            <BackTop visibilityHeight={80} target={() => this.EleRef} />
          </Demo>
        </div>
      </div>
    );
  }
}
