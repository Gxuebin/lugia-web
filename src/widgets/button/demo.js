/**
 *
 * create by ligx
 *
 * @flow
 */
import React from 'react';
import Button from './index';

export default () => {
  return <Button type="danger">hello</Button>;
};
