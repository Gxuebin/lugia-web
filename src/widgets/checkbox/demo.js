/**
 *
 * create by guorg
 *
 * @flow
 */
import React from 'react';
import styled from 'styled-components';
import Theme from '../theme';
import Widget from '../consts';
import CheckBox from './';
import { getBorder } from '@lugia/theme-css-hoc/lib/index';

const CheckBoxGroup = CheckBox.Group;
const CheckBoxButton = CheckBox.Button;
const Wrapper = styled.div`
  margin: 20px;
`;
const defaultValue = ['1', '2'];
const options = [
  { label: 'check1', value: '11', name: '1' },
  { label: 'check2', value: '22', name: '2' },
  { label: 'check3', value: '33', name: '3' },
];
const options2 = [
  { label: 'check1', value: '11', name: '1', disabled: true },
  { label: 'check2', value: '22', name: '2' },
  { label: 'check3', value: '33', name: '3' },
];
const handleChange = obj => {
  console.info(obj);
};
const view = {
  [Widget.Checkbox]: {
    color: 'red',
  },
  [Widget.CheckboxGroup]: {
    color: 'red',
    width: 100,
  },
  register: {
    width: 120,
    color: 'red',
    margin: 20,
  },
};
export class CheckboxDemo extends React.Component<any, any> {
  render() {
    const view = {
      [Widget.Checkbox]: {
        CheckboxWrap: {
          normal: {
            opacity: 1,
            color: 'blue',
            font: { fontWeight: 200, fontSize: 16 },
            width: 100,
            height: 50,
            margin: {
              top: 10,
              bottom: 10,
              left: 10,
              right: 10,
            },
            padding: {
              top: 10,
              bottom: 10,
              left: 10,
              right: 10,
            },
          },
          hover: {
            opacity: 0.6,
            color: 'red',
          },
          disabled: {
            opacity: 0.4,
            color: 'red',
          },
          active: {
            opacity: 1,
            color: 'green',
          },
        },
        CheckboxEdge: {
          normal: {
            background: { color: 'yellow' },
            border: getBorder({ color: '#56c22d', width: 2, style: 'solid' }, { radius: 2 }),
          },
          hover: {
            background: { color: '#56c22d' },
            border: getBorder({ color: 'yellow', width: 2, style: 'solid' }, { radius: 2 }),
          },
          disabled: {
            background: { color: 'orange' },
            border: getBorder({ color: 'red', width: 2, style: 'solid' }, { radius: 2 }),
          },
          active: {
            background: { color: '#3452c2' },
            border: getBorder({ color: 'red', width: 1, style: 'solid' }, { radius: 2 }),
          },
          cancel: {
            background: { color: 'pink' },
            border: getBorder({ color: 'red', width: 1, style: 'solid' }, { radius: 2 }),
          },
        },
        CheckboxChecked: {
          active: {
            background: { color: 'yellow' },
          },
          disabled: {
            background: { color: 'red' },
          },
          indeterminate: {
            background: { color: 'pink' },
          },
        },
      },
    };
    return (
      <div>
        <Wrapper>
          <Theme config={view}>
            <CheckBox>CheckBox</CheckBox>
          </Theme>
        </Wrapper>
        <Wrapper>
          <Theme config={view}>
            <CheckBox disabled>CheckBox</CheckBox>
          </Theme>
        </Wrapper>
        <Wrapper>
          <Theme config={view}>
            <CheckBox checked>CheckBox</CheckBox>
          </Theme>
        </Wrapper>
        <Wrapper>
          <Theme config={view}>
            <CheckBox cancel checked handleCancelItemClick={() => {}}>
              CheckBox
            </CheckBox>
          </Theme>
        </Wrapper>
        <Wrapper>
          <CheckBox cancel checked handleCancelItemClick={() => {}}>
            CheckBox
          </CheckBox>
        </Wrapper>
        <Wrapper>
          <Theme config={view}>
            <CheckBox indeterminate>CheckBox</CheckBox>
          </Theme>
        </Wrapper>
        <Wrapper>
          <CheckBox>CheckBox</CheckBox>
        </Wrapper>
        <Wrapper>
          <CheckBox checked onChange={handleChange}>
            CheckBox
          </CheckBox>
        </Wrapper>
        <Wrapper>
          <CheckBox defaultChecked>CheckBox</CheckBox>
        </Wrapper>
        <Wrapper>
          <CheckBox disabled>CheckBox</CheckBox>
        </Wrapper>
        <Wrapper>
          <CheckBox checked disabled>
            CheckBox
          </CheckBox>
        </Wrapper>
        <Wrapper>
          <CheckBox defaultChecked disabled>
            CheckBox
          </CheckBox>
        </Wrapper>
        <Wrapper>
          <CheckBox indeterminate>CheckBox</CheckBox>
        </Wrapper>
      </div>
    );
  }
}
export const CheckboxGroupDemo = class extends React.Component<any, any> {
  constructor(props: any) {
    super(props);
    this.state = {
      newValue: ['11', '44'],
      newDisplayValue: ['check4', 'check5'],
    };
  }

  handleChange = ({ newValue, newDisplayValue }: any): any => {
    console.info(newValue, newDisplayValue);
    this.setState({ newValue, newDisplayValue });
  };

  render() {
    const checkboxView = {
      [Widget.CheckboxGroup]: {
        Group: {
          normal: {
            width: 400,
            height: 100,
            opacity: 1,
            border: getBorder({ color: 'orange', width: 2, style: 'solid' }, { radius: 4 }),
            background: { color: 'red' },
            padding: {
              top: 10,
              bottom: 10,
              left: 10,
              right: 10,
            },
            margin: {
              top: 10,
              bottom: 10,
              left: 10,
              right: 10,
            },
          },
        },
        Checkbox: {
          CheckboxWrap: {
            normal: {
              opacity: 1,
              color: 'blue',
              font: { fontWeight: 200, fontSize: 16 },
              width: 100,
              height: 50,
              margin: {
                top: 10,
                bottom: 10,
                left: 10,
                right: 10,
              },
              padding: {
                top: 10,
                bottom: 10,
                left: 10,
                right: 10,
              },
            },
            hover: {
              opacity: 0.6,
              color: 'red',
            },
            disabled: {
              opacity: 0.4,
              color: 'red',
            },
            active: {
              opacity: 1,
              color: 'green',
            },
          },
          CheckboxEdge: {
            normal: {
              background: { color: 'yellow' },
              border: getBorder({ color: '#56c22d', width: 2, style: 'solid' }, { radius: 2 }),
            },
            hover: {
              background: { color: '#56c22d' },
              border: getBorder({ color: 'yellow', width: 2, style: 'solid' }, { radius: 2 }),
            },
            disabled: {
              background: { color: 'orange' },
              border: getBorder({ color: 'red', width: 2, style: 'solid' }, { radius: 2 }),
            },
            active: {
              background: { color: '#3452c2' },
              border: getBorder({ color: 'red', width: 1, style: 'solid' }, { radius: 2 }),
            },
            cancel: {
              background: { color: 'pink' },
              border: getBorder({ color: 'red', width: 1, style: 'solid' }, { radius: 2 }),
            },
          },
          CheckboxChecked: {
            active: {
              background: { color: 'yellow' },
            },
            disabled: {
              background: { color: 'red' },
            },
            indeterminate: {
              background: { color: 'pink' },
            },
          },
        },
      },
    };
    const checkbuttonView = {
      [Widget.CheckboxGroup]: {
        Group: {},
        CheckButton: {
          CheckButtonWrap: {
            normal: {
              width: 100,
              height: 50,
              opacity: 1,
              border: getBorder({ color: 'green', width: 2, style: 'solid' }, { radius: 4 }),
              background: { color: '#33f340' },
            },
            hover: {
              background: { color: 'green' },
              opacity: 0.6,
              border: getBorder({ color: 'pink', width: 2, style: 'solid' }, { radius: 4 }),
            },
            cancel: {
              border: getBorder({ color: 'green', width: 2, style: 'solid' }, { radius: 4 }),
              background: { color: '#8D13DE' },
            },
            active: {
              border: getBorder({ color: 'pink', width: 2, style: 'solid' }, { radius: 4 }),
              background: { color: 'red' },
            },
          },
          CheckButtonText: {
            normal: {
              color: 'blue',
              fontSize: 16,
              fontWeight: 400,
              padding: {
                top: 10,
                bottom: 10,
                left: 10,
                right: 10,
              },
            },
            hover: {
              color: 'red',
              fontSize: 12,
              fontWeight: 600,
            },
            active: {
              color: '#fff',
              fontSize: 18,
            },
          },
          Checked: {
            disabled: {
              opacity: 0.8,
              background: { color: '#b2a8ef' },
              border: getBorder({ color: 'yellow', width: 2, style: 'solid' }, { radius: 4 }),
              color: 'red',
            },
          },
          UnChecked: {
            disabled: {
              opacity: 0.4,
              background: { color: '#ccc' },
              border: getBorder({ color: 'green', width: 2, style: 'solid' }, { radius: 4 }),
              color: 'green',
            },
          },
        },
      },
    };
    return (
      <div>
        <Wrapper>
          <CheckBoxGroup onChange={this.handleChange} value={this.state.newValue}>
            <CheckBox value="11" disabled>
              CheckBox1
            </CheckBox>
            <CheckBox value="22">CheckBox2</CheckBox>
            <CheckBox value="33">CheckBox3</CheckBox>
          </CheckBoxGroup>
        </Wrapper>
        <Wrapper>
          <CheckBoxGroup onChange={this.handleChange}>
            <CheckBox value="11" disabled>
              CheckBox1
            </CheckBox>
            {123}
            <CheckBox value="22">CheckBox2</CheckBox>
            <CheckBox value="33">CheckBox3</CheckBox>
          </CheckBoxGroup>
        </Wrapper>
        <Wrapper>
          <Theme config={checkboxView}>
            <CheckBoxGroup
              onChange={handleChange}
              styles="vertical"
              data={options}
              defaultValue={['11']}
              displayField="label"
            />
          </Theme>
        </Wrapper>
        <Wrapper>
          <CheckBoxGroup
            onChange={this.handleChange}
            styles="vertical"
            data={options}
            value={['11', '44']}
            displayValue={['check4', 'check5']}
            displayField="label"
          />
        </Wrapper>
        <Wrapper>
          <CheckBoxGroup
            onChange={this.handleChange}
            styles="vertical"
            data={options}
            defaultValue={['11', '44']}
            displayValue={['check4', 'check5']}
            displayField="label"
          />
        </Wrapper>
        <Wrapper>
          <CheckBoxGroup childType="button" onChange={handleChange} value={defaultValue}>
            <CheckBoxButton value="1">CheckBox1</CheckBoxButton>
            <CheckBoxButton value="2">CheckBox2</CheckBoxButton>
            <CheckBoxButton value="3">CheckBox3</CheckBoxButton>
          </CheckBoxGroup>
        </Wrapper>
        <Wrapper>
          <Theme config={checkbuttonView}>
            <CheckBoxGroup childType="button" onChange={handleChange} defaultValue={defaultValue}>
              <CheckBoxButton value="1">CheckBox1</CheckBoxButton>
              <CheckBoxButton value="2">CheckBox2</CheckBoxButton>
              <CheckBoxButton value="3">CheckBox3</CheckBoxButton>
            </CheckBoxGroup>
          </Theme>
        </Wrapper>
        <Wrapper>
          <Theme config={checkbuttonView}>
            <CheckBoxGroup
              onChange={this.handleChange}
              data={options}
              value={this.state.newValue}
              displayValue={['check4', 'check5']}
              displayField="label"
              childType="button"
            />
          </Theme>
        </Wrapper>
        <Wrapper>
          <Theme config={checkbuttonView}>
            <CheckBoxGroup
              onChange={this.handleChange}
              data={options2}
              value={this.state.newValue}
              displayValue={['check4', 'check5']}
              displayField="label"
              childType="button"
            />
          </Theme>
        </Wrapper>
        <Wrapper>
          <Theme config={checkbuttonView}>
            <CheckBoxGroup
              onChange={this.handleChange}
              data={options2}
              value={'22'}
              displayValue={['check4', 'check5']}
              displayField="label"
              childType="button"
            />
          </Theme>
        </Wrapper>
        <Wrapper>
          <CheckBoxGroup
            styles="vertical"
            data={options}
            childType={'button'}
            defaultValue={['11', '44']}
            defaultDisplayValue={['check4', 'check5']}
            displayField="label"
          />
        </Wrapper>
      </div>
    );
  }
};
