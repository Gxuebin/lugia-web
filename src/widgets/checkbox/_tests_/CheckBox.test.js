/**
 * CheckBox 单元测试
 * create by guorg
 * @flow
 */
import React from 'react';
import chai from 'chai';

import 'jest-styled-components';
import Enzyme, { mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import CheckBox from '../';
import { CheckboxDemo } from '../demo';
import renderer from 'react-test-renderer';
import { delay } from '@lugia/react-test-utils';

const { expect: exp } = chai;

Enzyme.configure({ adapter: new Adapter() });

describe('CheckBox', () => {
  const getState = (target: any): Object => {
    const state = target
      .children()
      .at(0)
      .instance().state;
    return state;
  };
  const createTarget = (config: Object): any => {
    const target = mount(<CheckBox {...config} />);
    return target;
  };

  it('CheckBox CSS', () => {
    const target = <CheckboxDemo />;
    expect(renderer.create(target).toJSON()).toMatchSnapshot();
  });

  function onClick(target: any, result: boolean) {
    expect(getState(target).checked).toBe(true);
    const label = target.find('label');
    label.at(0).simulate('click');
    expect(getState(target).checked).toBe(result);
  }

  it('CheckBox: props: checked = true event: click ', () => {
    const config = { checked: true };
    const target = createTarget(config);
    onClick(target, true);
    onClick(target, true);
  });

  it('CheckBox props.checked mapTo state.checked ', () => {
    const config = { checked: true };
    const target = createTarget(config);
    expect(getState(target).checked).toBe(true);

    target.setProps({ checked: false });
    expect(getState(target).checked).toBe(false);

    target.setProps({ checked: true });
    expect(getState(target).checked).toBe(true);
  });

  it('CheckBox: props: defaultChecked = true event: click ', () => {
    const newConfig = { defaultChecked: true };
    const newTarget = createTarget(newConfig);
    onClick(newTarget, false);
  });

  it('CheckBox props: disabled  & onChange', () => {
    let triggerCnt = 0;
    let value = false;
    const handleChange = (_, val) => {
      triggerCnt++;
      value = val;
    };
    const config = { disabled: true, onChange: handleChange, value };

    const target = createTarget(config);

    function expectAfterClick(result: number, val: boolean) {
      const label = target.find('label');
      label.at(0).simulate('click');
      expect(triggerCnt).toBe(result);
      expect(value).toBe(val);
    }

    expectAfterClick(0, false);
    expectAfterClick(0, false);

    target.setProps({ disabled: false });

    expectAfterClick(1, true);
    expectAfterClick(2, false);

    target.setProps({ disabled: true });
    expectAfterClick(2, false);
  });
});
