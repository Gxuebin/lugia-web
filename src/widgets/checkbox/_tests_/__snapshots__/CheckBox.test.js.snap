// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBox CheckBox CSS 1`] = `
.c1 {
  font-size: 1.4rem;
  cursor: pointer;
  color: #333333;
  width: none;
  box-sizing: border-box;
  padding: 0;
  list-style: none;
  display: inline-block;
  position: relative;
  white-space: nowrap;
  display: inline-block;
  margin-right: 2.142857142857143em;
}

.c1:hover > span span {
  border: 1px solid #4d63ff;
}

.c7 {
  font-size: 1.4rem;
  cursor: not-allowed;
  color: #cccccc;
  width: none;
  box-sizing: border-box;
  padding: 0;
  list-style: none;
  display: inline-block;
  position: relative;
  white-space: nowrap;
  display: inline-block;
  margin-right: 2.142857142857143em;
}

.c7:hover > span span {
  border: 1px solid #e8e8e8;
}

.c12 {
  font-size: 1.4rem;
  cursor: pointer;
  color: #333333;
  width: 8.571428571428571em;
  box-sizing: border-box;
  padding: 0;
  list-style: none;
  display: inline-block;
  position: relative;
  white-space: nowrap;
  display: inline-block;
  margin-right: 2.142857142857143em;
  margin: 1.4285714285714286em;
}

.c12:hover > span span {
  border: 1px solid red;
}

.c2 {
  margin: 0;
  outline: none;
  line-height: 1;
  vertical-align: text-bottom;
  display: inline-block;
}

.c5 {
  padding-left: 0.7142857142857143em;
}

.c3 {
  position: absolute;
  left: 0;
  z-index: 1;
  cursor: pointer;
  opacity: 0;
  top: 0;
  bottom: 0;
  right: 0;
  width: 100%;
  height: 100%;
}

.c8 {
  position: absolute;
  left: 0;
  z-index: 1;
  cursor: not-allowed;
  opacity: 0;
  top: 0;
  bottom: 0;
  right: 0;
  width: 100%;
  height: 100%;
}

.c4 {
  position: relative;
  box-sizing: border-box;
  top: 0;
  left: 0;
  display: block;
  width: 1.2857142857142858em;
  height: 1.2857142857142858em;
  border-radius: 0.14285714285714288em;
  background-color: #fff;
  border: 1px solid #cccccc;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
}

.c4:hover {
  border: 1px solid #4d63ff;
}

.c4::after {
  position: absolute;
  box-sizing: border-box;
  left: 0.35714285714285715em;
  top: 0.14285714285714288em;
  width: 0.4285714285714286em;
  height: 0.7142857142857143em;
  -webkit-transform: rotate(45deg) scale(0);
  -ms-transform: rotate(45deg) scale(0);
  transform: rotate(45deg) scale(0);
  -webkit-transition: all .1s cubic-bezier(.71,-.46,.88,.6);
  transition: all .1s cubic-bezier(.71,-.46,.88,.6);
  display: table;
  border: 0.14285714285714288em solid #fff;
  border-top: 0;
  border-left: 0;
  content: ' ';
}

.c6 {
  position: relative;
  box-sizing: border-box;
  top: 0;
  left: 0;
  display: block;
  width: 1.2857142857142858em;
  height: 1.2857142857142858em;
  border-radius: 0.14285714285714288em;
  background-color: #4d63ff;
  border: 1px solid #4d63ff;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
}

.c6:hover {
  border: 1px solid #4d63ff;
}

.c6::after {
  position: absolute;
  box-sizing: border-box;
  left: 0.35714285714285715em;
  top: 0.14285714285714288em;
  width: 0.4285714285714286em;
  height: 0.7142857142857143em;
  -webkit-transform: rotate(45deg) scale(1);
  -ms-transform: rotate(45deg) scale(1);
  transform: rotate(45deg) scale(1);
  -webkit-transition: all .2s cubic-bezier(.71,-.46,.88,.6);
  transition: all .2s cubic-bezier(.71,-.46,.88,.6);
  display: table;
  border: 0.14285714285714288em solid #fff;
  border-top: 0;
  border-left: 0;
  content: ' ';
}

.c9 {
  position: relative;
  box-sizing: border-box;
  top: 0;
  left: 0;
  display: block;
  width: 1.2857142857142858em;
  height: 1.2857142857142858em;
  border-radius: 0.14285714285714288em;
  background-color: #f2f2f2;
  border: 1px solid #e8e8e8;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
}

.c9:hover {
  border: 1px solid #e8e8e8;
}

.c9::after {
  position: absolute;
  box-sizing: border-box;
  left: 0.35714285714285715em;
  top: 0.14285714285714288em;
  width: 0.4285714285714286em;
  height: 0.7142857142857143em;
  -webkit-transform: rotate(45deg) scale(0);
  -ms-transform: rotate(45deg) scale(0);
  transform: rotate(45deg) scale(0);
  -webkit-transition: all .1s cubic-bezier(.71,-.46,.88,.6);
  transition: all .1s cubic-bezier(.71,-.46,.88,.6);
  display: table;
  border: 0.14285714285714288em solid #cccccc;
  border-top: 0;
  border-left: 0;
  content: ' ';
}

.c10 {
  position: relative;
  box-sizing: border-box;
  top: 0;
  left: 0;
  display: block;
  width: 1.2857142857142858em;
  height: 1.2857142857142858em;
  border-radius: 0.14285714285714288em;
  background-color: #f2f2f2;
  border: 1px solid #e8e8e8;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
}

.c10:hover {
  border: 1px solid #e8e8e8;
}

.c10::after {
  position: absolute;
  box-sizing: border-box;
  left: 0.35714285714285715em;
  top: 0.14285714285714288em;
  width: 0.4285714285714286em;
  height: 0.7142857142857143em;
  -webkit-transform: rotate(45deg) scale(1);
  -ms-transform: rotate(45deg) scale(1);
  transform: rotate(45deg) scale(1);
  -webkit-transition: all .2s cubic-bezier(.71,-.46,.88,.6);
  transition: all .2s cubic-bezier(.71,-.46,.88,.6);
  display: table;
  border: 0.14285714285714288em solid #cccccc;
  border-top: 0;
  border-left: 0;
  content: ' ';
}

.c11 {
  position: relative;
  box-sizing: border-box;
  top: 0;
  left: 0;
  display: block;
  width: 1.2857142857142858em;
  height: 1.2857142857142858em;
  border-radius: 0.14285714285714288em;
  background-color: #4d63ff;
  border: 1px solid #4d63ff;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
}

.c11:hover {
  border: 1px solid #4d63ff;
}

.c11::after {
  position: absolute;
  box-sizing: border-box;
  left: 0.2142857142857143em;
  top: 0.5em;
  width: 0.7142857142857143em;
  height: 0.07142857142857144em;
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
  -webkit-transition: all .2s cubic-bezier(.03,.86,.56,.87);
  transition: all .2s cubic-bezier(.03,.86,.56,.87);
  display: table;
  border: 0.14285714285714288em solid #fff;
  border-top: 0;
  border-left: 0;
  content: ' ';
}

.c13 {
  position: relative;
  box-sizing: border-box;
  top: 0;
  left: 0;
  display: block;
  width: 1.2857142857142858em;
  height: 1.2857142857142858em;
  border-radius: 0.14285714285714288em;
  background-color: #fff;
  border: 1px solid #cccccc;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
}

.c13:hover {
  border: 1px solid red;
}

.c13::after {
  position: absolute;
  box-sizing: border-box;
  left: 0.35714285714285715em;
  top: 0.14285714285714288em;
  width: 0.4285714285714286em;
  height: 0.7142857142857143em;
  -webkit-transform: rotate(45deg) scale(0);
  -ms-transform: rotate(45deg) scale(0);
  transform: rotate(45deg) scale(0);
  -webkit-transition: all .1s cubic-bezier(.71,-.46,.88,.6);
  transition: all .1s cubic-bezier(.71,-.46,.88,.6);
  display: table;
  border: 0.14285714285714288em solid #fff;
  border-top: 0;
  border-left: 0;
  content: ' ';
}

.c14 {
  position: relative;
  box-sizing: border-box;
  top: 0;
  left: 0;
  display: block;
  width: 1.2857142857142858em;
  height: 1.2857142857142858em;
  border-radius: 0.14285714285714288em;
  background-color: red;
  border: 1px solid red;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
}

.c14:hover {
  border: 1px solid red;
}

.c14::after {
  position: absolute;
  box-sizing: border-box;
  left: 0.35714285714285715em;
  top: 0.14285714285714288em;
  width: 0.4285714285714286em;
  height: 0.7142857142857143em;
  -webkit-transform: rotate(45deg) scale(1);
  -ms-transform: rotate(45deg) scale(1);
  transform: rotate(45deg) scale(1);
  -webkit-transition: all .2s cubic-bezier(.71,-.46,.88,.6);
  transition: all .2s cubic-bezier(.71,-.46,.88,.6);
  display: table;
  border: 0.14285714285714288em solid #fff;
  border-top: 0;
  border-left: 0;
  content: ' ';
}

.c0 {
  margin: 20px;
}

<div>
  <div
    className="c0"
  >
    <label
      className="c1"
      onClick={[Function]}
    >
      <span
        className="c2"
        onClick={null}
      >
        <input
          className="c3"
          onBlur={[Function]}
          type="checkbox"
        />
        <span
          className="c4"
        />
      </span>
      <span
        className="c5"
      >
        CheckBox
      </span>
    </label>
  </div>
  <div
    className="c0"
  >
    <label
      className="c1"
      onClick={[Function]}
    >
      <span
        className="c2"
        onClick={null}
      >
        <input
          className="c3"
          onBlur={[Function]}
          type="checkbox"
        />
        <span
          checked={true}
          className="c6"
        />
      </span>
      <span
        className="c5"
      >
        CheckBox
      </span>
    </label>
  </div>
  <div
    className="c0"
  >
    <label
      className="c1"
      onClick={[Function]}
    >
      <span
        className="c2"
        onClick={null}
      >
        <input
          className="c3"
          onBlur={[Function]}
          type="checkbox"
        />
        <span
          checked={true}
          className="c6"
        />
      </span>
      <span
        className="c5"
      >
        CheckBox
      </span>
    </label>
  </div>
  <div
    className="c0"
  >
    <label
      className="c7"
      disabled={true}
      onClick={[Function]}
    >
      <span
        className="c2"
        onClick={null}
      >
        <input
          className="c8"
          disabled={true}
          onBlur={[Function]}
          type="checkbox"
        />
        <span
          className="c9"
          disabled={true}
        />
      </span>
      <span
        className="c5"
      >
        CheckBox
      </span>
    </label>
  </div>
  <div
    className="c0"
  >
    <label
      className="c7"
      disabled={true}
      onClick={[Function]}
    >
      <span
        className="c2"
        onClick={null}
      >
        <input
          className="c8"
          disabled={true}
          onBlur={[Function]}
          type="checkbox"
        />
        <span
          checked={true}
          className="c10"
          disabled={true}
        />
      </span>
      <span
        className="c5"
      >
        CheckBox
      </span>
    </label>
  </div>
  <div
    className="c0"
  >
    <label
      className="c7"
      disabled={true}
      onClick={[Function]}
    >
      <span
        className="c2"
        onClick={null}
      >
        <input
          className="c8"
          disabled={true}
          onBlur={[Function]}
          type="checkbox"
        />
        <span
          checked={true}
          className="c10"
          disabled={true}
        />
      </span>
      <span
        className="c5"
      >
        CheckBox
      </span>
    </label>
  </div>
  <div
    className="c0"
  >
    <label
      className="c1"
      onClick={[Function]}
    >
      <span
        className="c2"
        onClick={null}
      >
        <input
          className="c3"
          onBlur={[Function]}
          type="checkbox"
        />
        <span
          className="c11"
        />
      </span>
      <span
        className="c5"
      >
        CheckBox
      </span>
    </label>
  </div>
  <div
    className="c0"
  >
    <label
      className="c12"
      onClick={[Function]}
    >
      <span
        className="c2"
        onClick={null}
      >
        <input
          className="c3"
          onBlur={[Function]}
          type="checkbox"
        />
        <span
          className="c13"
        />
      </span>
      <span
        className="c5"
      >
        CheckBox
      </span>
    </label>
  </div>
  <div
    className="c0"
  >
    <label
      className="c12"
      onClick={[Function]}
    >
      <span
        className="c2"
        onClick={null}
      >
        <input
          className="c3"
          onBlur={[Function]}
          type="checkbox"
        />
        <span
          checked={true}
          className="c14"
        />
      </span>
      <span
        className="c5"
      >
        CheckBox
      </span>
    </label>
  </div>
</div>
`;
